### PR TITLE
fix: to correct value of pursArgs in findFlag in Path.hs

### PR DESCRIPTION
**Pull request by knight9999**
### Description of the change

This PR is to fix the bug that the user can not `spago run` command when he specifies the output directory different from `output`.

- The Issue I found

Support user wants to specify output directory, for example, `output2` directory.

The build can be done by the command
```
$ spago build -u "-o output2"
```

However, the run leads to error by the command.
```
$ spago run -u "-o output2"
```

- Reason and how to fix in this PR
 
This issue is because `findFlag` function in `Path.hs` has a problem.
If we use
```
$ spago run -u "-o output2"
```
, the `pursArgs` is `["-o output2"]`, not `["-o", "output2"]`.
 
Therefore `(x:y:xs)` can not match the `y = "output2"`.

### Checklist:

- [ ] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

**Comment by f-f** [comment]
Hi @knight9999, thanks for contributing the fix! :clap: 

This was sort of a known bug for a while, but not enough people stumbled on the edge cases for it to be annoying. In fact you can see I left a "TODO add tests" just a few lines above your fix:
https://github.com/purescript/spago/blob/7dcd30d810d29c2bf2ae61c7947d183255b1844e/src/Spago/Command/Path.hs#L46-L51

It would be great if you could add these tests as part of this PR, as it might help catch any other bugs we might be introducing (for example, there seems to be a regression in the current tests)

**Comment by knight9999** [comment]
Thanks @f-f for the valuable advice.
Now I have just added the test case and improved my PR to support `--output=something` case.

There are some test issue other than my PR,
Please test by using `Spec.hs` as

```
module Spec (spec) where

import Test.Hspec

import qualified Spago.Command.PathSpec as PathSpec

spec :: Spec
spec = do
  describe "PathSpec"     PathSpec.spec
```

instead of the origin one

```
{-# OPTIONS_GHC -F -pgmF hspec-discover -optF --module-name=Spec #-}
```

**Comment by knight9999** [comment]
Thanks @f-f for the valuable advice.
Now I have just added the test case and improved my PR to support `--output=something` case.

There are some test issue other than my PR,
Please test by using `Spec.hs` as

```
module Spec (spec) where

import Test.Hspec

import qualified Spago.Command.PathSpec as PathSpec

spec :: Spec
spec = do
  describe "PathSpec"     PathSpec.spec
```

instead of the origin one

```
{-# OPTIONS_GHC -F -pgmF hspec-discover -optF --module-name=Spec #-}
```

File: src/Spago/Command/Path.hs [hunk 1]
```diff
@@ -43,7 +43,7 @@ getOutputPath buildOpts = do
     Just path -> Text.unpack path
 
 
--- TODO tests:
+-- tests are followings: See test/Spago/Command/PathSpec.hs
```
Comment (f-f) [inline src/Spago/Command/Path.hs:46]:
```suggestion
```

**Review by f-f — APPROVED** [review]
I left a small comment but otherwise this looks great :clap:

**Comment by knight9999** [comment]
Hi @f-f . Thank you very much.

**Comment by f-f** [comment]
Thank you :slightly_smiling_face:
